ext/nnstreamer/tensor_filter/tensor_filter_support_cc.h

```diff
@@ -0,0 +1,83 @@
+/**
+ * GStreamer Tensor_filter, C++ Subplugin Support. (this is not a subplugin)
+ * Copyright (C) 2020 MyungJoo Ham <myungjoo.ham@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file	tensor_filter_support_cc.h
+ * @date	15 Jan 2020
+ * @brief	Base class for tensor_filter subplugins of C++ classes.
+ * @see		http://github.com/nnsuite/nnstreamer
+ * @author	MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ * @details
+ *    This is not a subplugin, but a helper for C++ subplugins.
+ *    If you want to write a wrapper for neural network frameworks
+ *    or a hardware adpator in C++, this is what you want.
+ *    If you want to attach a single C++ class/object as a filter
+ *    (a.k.a. custom filter) to a pipeline, use tensor_filter_cpp
+ *
+ * To Packagers:
+ *
+ * This is to be exposed with "nnstreamer-c++-dev"
+ */
+#ifndef __NNS_TENSOR_FILTER_CPP_SUBPLUGIN_SUPPORT_H__
+#define __NNS_TENSOR_FILTER_CPP_SUBPLUGIN_SUPPORT_H__
+#ifdef __cplusplus
+
+#include <nnstreamer_plugin_api_filter.h>
+
+/**
+ * @brief The base class for C++ subplugins. Derive this to write one.
+ *
+ * @detail The subplugin writers (derived class writer) are supposed to
+ *        write virtual functions for their subplugins. They may
+ *        add their own data structures as well in the derived class.
+ *         Unlike C-version, constructing an object will automatically
+ *        register(probe) the subplugin for nnstreamer.
+ *         Optional virtual functions (non pure virtual functions) may
+ *        be kept un-overriden if you don't support such.
+ *         For getInput/Output and setInput, return -EINVAL if you don't
+ *        support it.
+ */
+class tensor_filter_subplugin {
+  private: /** subclassses are not supposed to access this. */
+    const uint32_t validity; /**< Sanity check */
+  public: /** subclasses may access this. */
+    const GstTensorFilterProperties *prop;
+
+    const char *name; /**< The name of this subplugin. E.g., "edgetpu" */
+    bool allow_in_place;
+    bool allocate_in_invoke;
+    bool run_without_model;
+    bool verify_model_path;
+
+    virtual int invoke_NN (const GstTensorMemory * input, GstTensorMemory * output) = 0;
+    virtual int getInputDimension (GstTensorsInfo * info) = 0;
+    virtual int getOutputDimension (GstTensorsInfo * info) = 0;
```

Comment:
How about adding some comments for `getInput/outputDimension` as well (like in `tensor_filter_cpp.h`) ?
Because they can be NULL if `setInputDimension` is defined.

Reply:
It'd be a bit different from tensor_filter_cpp.h, but yes I need comments.

Anyway, I'll update them from a pure virtual to virtual so that "not deriving the function" means that the function is not supported.